susemanager-utils/susemanager-sls/src/grains/public_cloud.py

```diff
@@ -111,3 +111,9 @@ def instance_id():
     if INSTANCE_ID:
         ret['instance_id'] = INSTANCE_ID
     return ret
+
+def is_payg_instance():
+    ret = {}
+    if os.path.isfile('/usr/sbin/registercloudguest'):
+        ret['is_payg_instance'] = True
```

Comment:
Is the existance of this file sufficient to detect a PAYG instance? Does it work for all clouds?

Reply:
This is actually what "SUSEConnect" is currently doing to differentiate between BYOS and PAYG instances: https://github.com/SUSE/connect/commit/6f26802c4b5e23c8133e534353b6067d59e6da8e#diff-ce4a8c41d114461bdba2a49aef6b34a2R55

Reply:
ok